### PR TITLE
Test better error messages when Makefiles fail

### PR DESCRIPTION
**Pull request by ktbarrett**
We get error messages when tests fail in cocotb, but not if the Makefile fails. This should help.

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -188,7 +188,7 @@ def dev_test_sim(
         coverage_file.unlink()
 
     session.log(f"Running 'make test' against a simulator {config_str}")
-    session.run("make", "test", external=True, env=env)
+    session.run("make", "-k", "test", external=True, env=env)
```
Comment (cmarqu) [inline noxfile.py:191]:
```suggestion
    session.run("make", "--keep-going", "test", external=True, env=env)
```
maybe? We require GNU make in the docs, so it should be available everywhere.

File: Makefile [hunk 2]
```diff
@@ -40,9 +40,9 @@ clean:
 
 .PHONY: do_tests
 do_tests::
-	$(MAKE) -k -C tests
+	$(MAKE) -C tests
```
Comment (cmarqu) [inline Makefile:43]:
```suggestion
	$(MAKE) --directory=tests
```

File: Makefile [hunk 3]
```diff
@@ -40,9 +40,9 @@ clean:
...
 do_tests::
-	$(MAKE) -k -C examples
+	$(MAKE) -C examples
```
Comment (cmarqu) [inline Makefile:45]:
```suggestion
	$(MAKE) --directory=examples
```

**Comment by ktbarrett** [comment]
[Errors](https://github.com/cocotb/cocotb/actions/runs/8767742529).

**Comment by ktbarrett** [comment]
I know the fully spelled names are clearer, but the short ones are standard a pretty well understood. Also I'm going to rm the Makefiles and do everything with nox soon.

**Review by cmarqu — APPROVED** [review]

**Pull request merged**